autoreject/utils.py

```diff
@@ -165,33 +165,20 @@ def _pprint(params, offset=0, printer=repr):
     return lines
 
 
-def _pbar(iterable, desc, leave=True, position=None, verbose='progressbar'):
-
-    verbose = False if verbose == 0 else verbose
-    if verbose is not False and \
-            verbose not in ['progressbar', 'tqdm', 'tqdm_notebook']:
-        raise ValueError('verbose must be one of {progressbar,'
-                         'tqdm, tqdm_notebook, False}. Got %s' % verbose)
-
-    try:
-        from tqdm import tqdm
-        verbose = 'tqdm'
-    except ImportError:
-        pass
-
-    if verbose == 'progressbar':
-        from mne.utils import ProgressBar
-        pbar = ProgressBar(iterable, mesg=desc)
-    # XXX: remove the tqdm option after a few releases of MNE since it
-    # natively supported by the MNE progressbar
-    elif verbose == 'tqdm':
-        pbar = tqdm(iterable, desc=desc, leave=leave, position=position,
-                    dynamic_ncols=True)
-    elif verbose == 'tqdm_notebook':
-        from tqdm import tqdm_notebook
-        pbar = tqdm_notebook(iterable, desc=desc, leave=leave,
-                             position=position, dynamic_ncols=True)
-    elif verbose is False:
+def _pbar(iterable, desc, verbose=True, **kwargs):
```

Comment:
So the `leave` and `position` options were for nested loops in autoreject. See my [issue on tqdm](https://github.com/tqdm/tqdm/issues/312#issuecomment-261098464). Basically, we want cross-validation fold as the outer loop and `n_interp` nested inside it. It was working before for `tqdm` backend but now it would be broken. Any chance we could get this to work with MNE?

Reply:
These options are covered in `**kwargs` and I'm getting the behavior described in the issue you linked (screenshots below). The `leave` and `position` variables are passed as arguments when this functionality is needed ([`n_interp`](https://github.com/autoreject/autoreject/blob/a60ac1a7a729429d34b075f6644d32604236dc69/autoreject/autoreject.py#L772-L775) and [`Fold`](https://github.com/autoreject/autoreject/blob/a60ac1a7a729429d34b075f6644d32604236dc69/autoreject/autoreject.py#L805-L806))

<img width="457" alt="1" src="https://user-images.githubusercontent.com/16963003/130523750-f483c353-767a-46ee-aba6-96298f5e0e85.png">
<img width="457" alt="2" src="https://user-images.githubusercontent.com/16963003/130523747-5b7d36d0-0214-4095-b497-3730e81cf9b5.png">
<img width="456" alt="3" src="https://user-images.githubusercontent.com/16963003/130523742-74dec602-766f-46bc-90e5-729c42e14dea.png">
<img width="454" alt="4" src="https://user-images.githubusercontent.com/16963003/130523736-83511666-d1a8-4dfb-b49a-d31b553065f3.png">

Reply:
If you like, I can restore the default arguments (`leave=True, position=None`) to the [`_pbar`](https://github.com/autoreject/autoreject/blob/91dee4f7a8d4d76ab3bc2c14c3ff91416dcb513f/autoreject/utils.py#L168)? Sorry if this is so much iteration! It's my first time contributing to open source!

Reply:
no worries! I think I noticed this now but looks like `tqdm` has changed (or broken?) since the last time I posted that question. It looks like `leave=True` always creates a new line but this was not the case before. It would leave a trace but not create a new line. Try this:

```py
from tqdm import trange
from time import sleep

for i in trange(10, desc='1st loop', leave=True, position=1):
    for j in trange(100, desc='2nd loop', leave=True, position=2):
        sleep(0.01)
    for k in trange(100, desc='3nd loop', leave=True, position=3):
        sleep(0.01)
```

vs 

```py
from tqdm import trange
from time import sleep

for i in trange(10, desc='1st loop', leave=True, position=1):
    for j in trange(100, desc='2nd loop', leave=False, position=2):
        sleep(0.01)
    for k in trange(100, desc='3nd loop', leave=False, position=3):
        sleep(0.01)
```

Neither is the desired behavior that was working before but the second one is closer to what I wanted I suppose (except that the trace should not be erased at the end of the loop)